source/slang/slang-emit-c-like.cpp

```diff
@@ -1105,7 +1105,23 @@ String CLikeSourceEmitter::getName(IRInst* inst)
     String name;
     if(!m_mapInstToName.tryGetValue(inst, name))
     {
-        name = generateName(inst);
+        // unmangle names, when emitting header
+        if (isEmmitingHeader()) {
+            if (auto nameHintDecor = inst->findDecoration<IRNameHintDecoration>()) 
+            {
+                StringBuilder sb;
+                for (auto c : nameHintDecor->getName()) {
+                    if (c == '.')
+                        sb.append('_');
+                    else {
+                        sb.append(c);
+                    }
+                }
+                name = sb.produceString();
```

Comment:
Wouldn't this conflict with another name that actually uses '_' as a part of its name?
You may want to check how "__extern_cpp" avoids the name mangling.

Reply:
But I would like to keep namespaces somehow, if possible.

Reply:
I changed this to use `appendScrubbedName`, but it still does basically the same - replaces dot with underscore.

Reply:
This shouldn't be necessary because `generateName(inst)` will already call `appendScrubbedName` for you to generate a valid name for the target. Why do we need a separate branch here?

Reply:
I suggest using __extenr_cpp in user code and remove this special case logic.

Reply:
`generateName(inst)` generates names like `_S1,  _S2` or `something_0`, `something_1` I wanted to get original names from Slang without any suffixes or changes.

Reply:
Main downside of using `__extern_cpp` is that it does not support namespaces. Unfortunately I think I would really need to drop this, otherwise there is a name mismatch between header and source code - for some reason it worked, but it is not a clear solution. Any ideas how to have `__extern_cpp` that supports namespaces? Because for example, if I want to have CPU/CUDA code (that switches at runtime if cuda is available), I would have name clashes - and namespaces could solve this for me (if they are different for CPU and CUDA sources).

Reply:
How about this:
we allow `__extern_cpp` on `namespace`s as well. And `__extern_cpp` on a decl means that the decl gets a mangled name that is in the form of: `parentNamespaceNameThatHasExternCppModifier_DeclName`.

So:
```
__extern_cpp namespace A
{
         namespace B {
              __extern_cpp struct C;
        }
}
Will have the mangled name `A_C`.
```

Reply:
Yes this would be nice, thanks.

Unfortunately it probably won't fully solve my problem with linking together CPU and CUDA code.